<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from tripteronBody.urdf.xacro       | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="tripteron">
  <!-- Util class-->
  <!--                         numbers:                    -->
  <!-- 0.1524 = 6in    -->
  <!-- 0.1016 = 3in   -->
  <!-- 0.0254 = 1in    -->
  <!-- 0.0127 = 0.5in -->
  <!-- *************** Materials / Colors **************** -->
  <material name="white">
    <color rgba="1.00 1.00 1.00 1"/>
  </material>
  <material name="gray">
    <color rgba="0.60 0.60 0.60 1"/>
  </material>
  <material name="table">
    <color rgba="0.85 0.77 0.77 1"/>
  </material>
  <material name="black">
    <color rgba="0.00 0.00 0.00 1"/>
  </material>
  <material name="red">
    <color rgba="1.00 0.00 0.00 1"/>
  </material>
  <material name="green">
    <color rgba="0.00 1.00 0.00 1"/>
  </material>
  <material name="blue">
    <color rgba="0.00 0.00 1.00 1"/>
  </material>
  <material name="cyan">
    <color rgba="0.00 1.00 1.00 1"/>
  </material>
  <material name="magenta">
    <color rgba="1.00 0.00 1.00 1"/>
  </material>
  <material name="yellow">
    <color rgba="1.00 1.00 0.00 1"/>
  </material>
  <material name="orange">
    <color rgba="1.00 0.65 0.00 1"/>
  </material>
  <!-- URDF relations to IRL are in meters-->
  <link name="platform">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <!-- Physical Center of robot Endeffector-->
      <geometry>
        <box size="0.1524 0.1524 0.1016"/>
      </geometry>
      <material name="magenta"/>
    </visual>
  </link>
  <!-- ******************** ARM ONE ******************* -->
  <!-- Base_link to Upper arm  -->
  <joint name="theta1" type="continuous">
    <parent link="platform"/>
    <child link="upperArm_one"/>
    <origin rpy="-1.5707963267948966 -0.7853981633974483 0" xyz="0.05 -0.05 -0.0508"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="upperArm_one">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.0762"/>
      <geometry>
        <cylinder length="0.1524" radius="0.0127"/>
      </geometry>
      <material name="cyan"/>
    </visual>
  </link>
  <!-- Upper arm to Lower Arm -->
  <joint name="theta2" type="continuous">
    <parent link="upperArm_one"/>
    <child link="lowerArm_one"/>
    <origin rpy="-0.7853981633974483 0 0" xyz="0 0 -0.1524"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="lowerArm_one">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0762"/>
      <geometry>
        <cylinder length="0.1524" radius="0.0127"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>
  <!-- Lower Arm to rail attachment -->
  <joint name="theta3" type="fixed">
    <parent link="lowerArm_one"/>
    <child link="railAttach_one"/>
    <origin rpy="0 0 0" xyz="0 0 0.1524"/>
  </joint>
  <link name="railAttach_one"/>
  <!-- ******************** ARM TWO ******************* -->
  <!-- Base_link to Upper arm  -->
  <joint name="theta4" type="continuous">
    <parent link="platform"/>
    <child link="upperArm_two"/>
    <origin rpy="-1.5707963267948966 -0.7853981633974483 0" xyz="0.05 0.05 -0.0508"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="upperArm_two">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0762"/>
      <geometry>
        <cylinder length="0.1524" radius="0.0127"/>
      </geometry>
      <material name="cyan"/>
    </visual>
  </link>
  <!-- Upper arm to Lower Arm -->
  <joint name="theta5" type="continuous">
    <parent link="upperArm_two"/>
    <child link="lowerArm_two"/>
    <origin rpy="-2.356194490192345 0 0" xyz="0 0 0.1524"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="lowerArm_two">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0762"/>
      <geometry>
        <cylinder length="0.1524" radius="0.0127"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>
  <!-- Lower Arm to rail attachment -->
  <joint name="theta6" type="continuous">
    <parent link="lowerArm_two"/>
    <child link="railAttach_two"/>
    <origin rpy="0 0 0" xyz="0 0 0.1524"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="railAttach_two"/>
  <!-- ******************** ARM THREE ******************* -->
  <!-- Base_link to Upper arm  -->
  <joint name="theta7" type="continuous">
    <parent link="platform"/>
    <child link="upperArm_three"/>
    <origin rpy="1.0471975511965976 0.7853981633974483 -0.7853981633974483" xyz="-0.05 -0.05 -0.0508"/>
    <!-- Come back later to mess with the yaw  -${pi/4} -->
    <axis xyz="1 0 0"/>
  </joint>
  <link name="upperArm_three">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0762"/>
      <geometry>
        <cylinder length="0.1524" radius="0.0127"/>
      </geometry>
      <material name="cyan"/>
    </visual>
  </link>
  <!-- Upper arm to Lower Arm -->
  <joint name="theta8" type="continuous">
    <parent link="upperArm_three"/>
    <child link="lowerArm_three"/>
    <origin rpy="-2.0943951023931953 0 3.141592653589793" xyz="0 0 0.1524"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="lowerArm_three">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0762"/>
      <geometry>
        <cylinder length="0.1524" radius="0.0127"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>
  <!-- Lower Arm to rail attachment -->
  <joint name="theta9" type="continuous">
    <parent link="lowerArm_three"/>
    <child link="railAttach_three"/>
    <origin rpy="0 0 0" xyz="0 0 0.1524"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="railAttach_three"/>
  <!-- ******************** ARM FOUR ******************* -->
  <!-- Base_link to Upper arm  -->
  <joint name="theta10" type="continuous">
    <parent link="platform"/>
    <child link="upperArm_four"/>
    <origin rpy="-1.0471975511965976 0.7853981633974483 0.7853981633974483" xyz="-0.05 0.05 -0.0508"/>
    <!-- Come back later to mess with the yaw  -0.7854 -->
    <axis xyz="1 0 0"/>
  </joint>
  <link name="upperArm_four">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0762"/>
      <geometry>
        <cylinder length="0.1524" radius="0.0127"/>
      </geometry>
      <material name="cyan"/>
    </visual>
  </link>
  <!-- Upper arm to Lower Arm -->
  <joint name="theta11" type="continuous">
    <parent link="upperArm_four"/>
    <child link="lowerArm_four"/>
    <origin rpy="2.0943951023931953 0 3.141592653589793" xyz="0 0 0.1524"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="lowerArm_four">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0762"/>
      <geometry>
        <cylinder length="0.1524" radius="0.0127"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>
  <!-- Lower Arm to rail attachment -->
  <joint name="theta12" type="continuous">
    <parent link="lowerArm_four"/>
    <child link="railAttach_four"/>
    <origin rpy="0 0 0" xyz="0 0 0.1524"/>
    <!-- rpy="${pi/3} ${3*pi/4} -${pi/2}"  |${pi/4} -${pi/2} 0-->
    <axis xyz="1 0 0"/>
  </joint>
  <link name="railAttach_four"/>
</robot>
